_posts/2020-10-23-deconvolution.md

```diff
@@ -0,0 +1,387 @@
+---
+layout: post
+title: Image Analysis Redux
+tagline: Dask + CuPy + RL
+author: John Kirkham (NVIDIA) and Ben Zaitlen (NVIDIA)
+theme: twitter
+---
+{% include JB/setup %}
+
+Summary
+-------
+
+
+Image Analysis Redux
+--------------------
+
+[Last year](https://blog.dask.org/2019/08/09/image-itk) we experimented with Dask/ITK/Scikit-Image
+to perform large scale image analysis on a stack of 3D images.  Specifically, we looked at `deconvolution`,
+a common method to _deblur_ images.  Now, a year later, we return to these experiments with a better
+understanding of how Dask and CuPy can interact, enhanced serialization methods, and support from the
+open-source community.
+
+Previously we used the [Richardson Lucy (RL)](https://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution) deconvolution
+algorithm from ITK and [Scikit-Image](https://github.com/scikit-image/scikit-image/blob/master/skimage/restoration/deconvolution.py#L329).
+We left off at theorizing how GPUs could potentially help accelerate these workflows.  Starting with Scikit-Image's
+implementation, we naively tried replacing `scipy.signal.convolve` calls with `cupyx.scipy.ndimage.convolve`, and while
+performance improved, it did not improve _significantly_ -- that is, we did not get the 100X speed we were looking for.
+
+Often, in working with images, we hit performance roadblocks in real space where data represents position.  When this happens,
```

Comment:
I know this is still WIP and subject to change, but I think this paragraph is a bit confusing for someone not already familiar with the concepts. We can talk about this at a more appropriate time though.

Reply:
I agree.  I was trying to find some language for those unfamiliar with FFTs to help understand our motivations.  Do you think we should remove entirely or try to re-work?

Reply:
I tried giving an (admittedly hand wavy) explanation that didn't rely on understanding FFTs as much. Please let us know what you think :)